Node.js: add notes for Next.js / Nest.js framework (#23503)
DataDog/documentation: content/en/tracing/trace_collection/compatibility/nodejs.md

| [paperplane][19]        | `>=2.3`  | Fully supported | Not supported in [serverless-mode][20]     |
| [restify][21]           | `>=3`    | Fully supported |                                            |

#### Complex framework usage

Some modern complex Node.js frameworks, such as Next.js and Nest.js, provide their own entry-point into an application. Instead of running something like `node app.js` they instead recommend running something like `next start`. In this case the entry point changes from a local application file, `app.js`, to a file that ships in the framework package. In this case it doesn't matter how early in your application code you load the tracer. By the time your application code runs the framework has already loaded modules that should have already been instrumented by the tracer. A different approach needs to be taken to load the tracer before the framework loads.

> The tracer can be loaded before any framework code by modifying the command used to start the application. One way to do this is to prefix all commands you run with an environment variable:
> 
> ```sh
> NODE_OPTIONS='--require dd-trace/init' npm start
> ```

Reviewer: ```suggestion
To load the tracer before the framework, use one of the following methods:

Prefix all commands you run with an environment variable:

```shell
NODE_OPTIONS='--require dd-trace/init' npm start
```
```

> Another way to achieve this is to modify the _package.json_ file if you typically start an application with npm or yarn run scripts:

Reviewer 2: Should package.json be in backticks instead of underscores?

> Another way to achieve this is to modify the `package.json` file if you typically start an application with npm or yarn run scripts:

Reviewer: ```suggestion
Or, modify the `package.json` file if you typically start an application with npm or yarn run scripts:
```

```plain
    // existing command
    "start": "next start",

    // suggested command
    "start": "node --require dd-trace/initialize ./node_modules/next start",
    "start": "NODE_OPTIONS='--require dd-trace/initialize' ./node_modules/next start",
```

> This example assumes you're using the Next.js framework but the same pattern applies to Nest.js and likely other frameworks as well. Feel free to adapt it for your situation. Either command should work however note that using `NODE_OPTIONS` should also apply to any child Node.js processes as well.

Reviewer: ```suggestion
**Note**: The previous examples use Next.js, but the same approach applies to other frameworks with custom entry points, such as Nest.js. Adapt the commands to fit your specific framework and setup. Either command should work, but using `NODE_OPTIONS`  also applies to any child Node.js processes.
```
